source/mainnet/net/guides/wallet-sdk/wallet-sdk-credential-deployment.rst

```diff
@@ -89,7 +89,93 @@ The following example demonstrates how a credential deployment transaction is cr
 
         Kotlin (Android)
 
-        TODO Write the Kotlin example.
+        .. code-block:: Kotlin
+
+            import cash.z.ecc.android.bip39.Mnemonics
+            import cash.z.ecc.android.bip39.toSeed
+            import com.concordium.example.wallet.services.ConcordiumClientService
+            import com.concordium.example.wallet.services.ConcordiumWalletProxyService
+            import com.concordium.sdk.crypto.wallet.ConcordiumHdWallet
+            import com.concordium.sdk.crypto.wallet.Credential
+            import com.concordium.sdk.crypto.wallet.Network
+            import com.concordium.sdk.crypto.wallet.UnsignedCredentialInput
+            import com.concordium.sdk.crypto.wallet.credential.CredentialDeploymentDetails
+            import com.concordium.sdk.crypto.wallet.credential.UnsignedCredentialDeploymentInfoWithRandomness
+            import com.concordium.sdk.requests.BlockQuery
+            import com.concordium.sdk.responses.accountinfo.credential.AttributeType
+            import com.concordium.sdk.transactions.CredentialPublicKeys
+            import com.concordium.sdk.transactions.Expiry
+            import com.concordium.sdk.transactions.Index
+            import java.util.Collections
+            import java.util.EnumMap
+
+            fun createCredentialDeploymentTransaction(): CredentialDeploymentDetails {
+                // The identity object created previously. See the identity creation section.
+                val identity: IdentityObject = ...
+
+                // The identity provider that was used for creating the identity.
```

Comment:
This comment doesn't quite make sense.
(Also maybe rename the variable to identityProviderIndex? :thinking:)

Reply:
I have renamed it.